### DISCUSSION TITLE
Various possible running changes & ideas for minor bugfixes

### DISCUSSION THREAD

**Discussion by exaexa** [LCSB-BioCore/COBREXA.jl#408] [Ideas] [answered]
converted from @stelmo's issue:

More small improvements, don't address these, I will when I am done adding to this list. However, feel free to comment on the changes, maybe I am wrong...

- [x] `Reaction` has fields called `lb` and `ub`, change these to `lower_bound` and `upper_bound` to match the more verbose style we have adopted. ⚠️ this would be a breaking change
- [x] make a `set_objective!` function
- [x] `change_constraint` let it be possible to use kwargs to only change a specific bound
- [x] `flux_summary` should not error when `nothing` is input, but rather display a message saying the solver didn't return a solution
- [x] `metabolite_fluxes` why ignore zero flux metabolites? change the function to drop the dependence on `_constants.tolerance`
- [x] change documentation for `change_objective` to reflect the fact that MOI is not necessarily imported e.g. `change_objective(...; sense=MOI.MIN_SENSE)` will error if `JuMP` is not imported, rather have the docstring read: `change_objective(...; sense=COBREXA.MOI.MIN_SENSE)`
- [x] `flux_summary` should not error when there is no biomass or import or export reactions
- [x] use `@kwdef` (see #566)

**Comment by stelmo** [comment, marked as answer]
I think this discussion is basically done, only kwdefs is really left. That could be something that goes into 2.0

**Reply by exaexa** [reply]
true, certainly will come up when redoing StdModel
